src/core/server/metrics/metrics_service.ts

```diff
@@ -30,13 +34,17 @@ export class MetricsService
   private collectInterval?: NodeJS.Timeout;
   private metrics$ = new ReplaySubject<OpsMetrics>(1);
   private service?: InternalMetricsServiceSetup;
+  private subscription: Subscription = new Subscription();
```

Comment:
No need to create the subscription (or we'll have to close it before reassigning the variable)
```suggestion
  private subscription: Subscription;
```

Then, in the stop method, we can access the subscription with an optional chain:

```typescript
this.subscription?.unsubscribe();
```